pkg/apis/kubernetes/v1/cluster_types.go

```diff
@@ -31,6 +32,23 @@ type Cluster struct {
 	// Requires Service VMs.
 	EnableLBaaS *bool `json:"enable_lbaas,omitempty"`
 
+	// Identifier of an existing internal v4 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageInternalIPv4Prefix` is true.
+	InternalIPv4Prefix *common.PartialResource `json:"internal_ipv4_prefix,omitempty"`
+	// Identifier of an existing external v4 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageExternalIPv4Prefix` is true.
+	ExternalIPv4Prefix *common.PartialResource `json:"external_ipv4_prefix,omitempty"`
+	// Identifier of an existing external v6 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageExternalIPv6Prefix` is true.
+	ExternalIPv6Prefix *common.PartialResource `json:"external_ipv6_prefix,omitempty"`
```

Comment:
Change docs, these attributes are not only useful when creating a cluster but you will also get the automatically created prefixes after creating a cluster with managed prefixes

Also reword the second part to ensure users set the attributes below to false if they want to supply their own prefixes on cluster creation - "Creating clusters in existing prefixes requires the Manage..Prefix attribute below to be set to false"

Reply:
Maybe something like
```golang
	// Identifier of an internal v4 prefix (to be) assigned to the cluster. If ManageInternalIPv4Prefix
	// is set to false, the value of this field is used to create the cluster. Otherwise a new prefix
	// will be created automatically. Updates to actual value after Create, Get & Update.
	InternalIPv4Prefix *common.PartialResource `json:"internal_ipv4_prefix,omitempty"`
```

Reply:
Good, but I don't like "the value of this field" as we can be more specific and I don't like the last sentence, what about those?

`[...] the Prefix given in this field is used when creating the cluster, otherwise a new [...]`

`Updates to actual value after Create, Get & Update.` -> `The API will always return the Prefix for the Cluster, when ManageInternalIPv4Prefix is true, this will be the Prefix that was created automatically.`